dot/core/service.go

```diff
@@ -373,14 +375,14 @@ func (s *Service) handleChainReorg(prev, curr common.Hash) error {
 
 		for _, ext := range *body {
 			logger.Tracef("validating transaction on re-org chain for extrinsic %s", ext)
-			encExt, err := scale.Marshal(ext)
+			decExt := &ctypes.Extrinsic{}
+			decoder := cscale.NewDecoder(bytes.NewReader(ext))
+			_, err := decoder.DecodeUintCompact()
```

Comment:
Why the change? 🤔

Reply:
I believe we just did not need to marshal here. Then we were using our own type that was a wrapper for cscale and calling DecodeVersion(). Since this function is the only place its used, i moved the logic here to make it more clear (I spent a lot of time trying to figure out what was going on) and deleted the wrapper type.